src/hooks/useConnectionForm.ts

```diff
@@ -340,7 +353,10 @@ export const useConnectionForm = (validationSchema: ZodObject<ZodRawShape>, mode
 		editConnection(connectionId!, connectionIntegrationName);
 	};
 
-	const handleOAuth = async (oauthConnectionId: string, integrationName: keyof typeof Integrations) => {
+	const handleOAuth = async (
+		oauthConnectionId: string,
+		integrationName: keyof typeof Integrations | typeof defaultMicrosoftConnectionName
+	) => {
```

Comment:
Does it looks better?

```suggestion
	type IntegrationNameType = keyof typeof Integrations | typeof defaultMicrosoftConnectionName;
	const handleOAuth = async (oauthConnectionId: string, integrationName: IntegrationNameType) => {
```

Reply:
I refactored and removed it